src/LottoConsoleGame.js

```diff
@@ -0,0 +1,123 @@
+import { LOTTO, RANKING_THRESHOLD, GAME_COMMAND } from './constants';
+import lottoGameCalculator from './domain/lottoGameCalculator';
+import lottoGameValidator from './domain/lottoGameValidator';
+import Lotto from './domain/models/Lotto';
+import generateRandomNumber from './utils/generateRandomNumber';
+import Interface from './view/Interface';
+import outputView from './view/outputView';
+
+class LottoConsoleGame {
+  #io;
+  #lottos;
+
+  constructor() {
+    this.#io = new Interface();
+    this.#lottos = [];
+  }
+
+  async play() {
+    const pruchaseAmount = await this.readPurchaseAmount();
+    this.buyLottos(pruchaseAmount);
+    this.printLottos();
+
+    const winningNumbers = await this.readWinningNumbers();
+    const bonusNumber = await this.readBonusNumber(winningNumbers);
+    this.printStatistics(pruchaseAmount, this.makeRankings(winningNumbers, bonusNumber));
+
+    this.decideReplay(await this.readGameCommand());
+  }
+
+  buyLottos(purchaseAmount) {
+    while (this.#lottos.length < purchaseAmount / LOTTO.price) {
+      this.#lottos.push(this.buyLotto());
+    }
+  }
+
+  buyLotto() {
+    const randomNumbers = [];
+    while (randomNumbers.length < LOTTO.numbersLength) {
+      const randomNumber = generateRandomNumber(LOTTO.minNumber, LOTTO.maxNumber);
+      if (!randomNumbers.includes(randomNumber)) randomNumbers.push(randomNumber);
+    }
```

Comment:
while 문으로 계속 체크하고 있군요.
while loop는 통과하겠지만, 다른 방법은 없을까요?

힌트를 드리자면, 카드 셔플 로직을 어떻게 구현할지 생각해면 감이 올거에요.

Reply:
생각해 본 적 없었는데 질문해주셔서 감사합니다!
아래와 같이 구현할 수 있을 거 같습니다.
1. 1부터 45의 배열을 준비합니다.
2. 랜덤하게 정렬하고 6개만 자릅니다.

```suggestion
    const randomNumbers = Array.from({ length: LOTTO.maxNumber - 1 }, (_, i) => i + 1)
      .sort(() => Math.random() - 0.5)
      .slice(0, LOTTO.numbersLength);
```